### PR TITLE
Migrate hideChaff() from Blockly to WorkspaceSvg

### PR DESCRIPTION
**Pull request by gonfunko**
## The basics

- [x] I branched from `goog_module`
- [x] My pull request is against `goog_module`
- [x] My code follows the [style guide](
      https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] My code is presented in the form suggested in the [module
      conversion guide](https://github.com/google/blockly/issues/5026)
- [x] I have run `npm test`.

### Resolves

Part of #5208

### Proposed Changes

Migrates hideChaff() from Blockly to WorkspaceSvg. This slightly changes the semantics of the method; before, it would hide global modal items (eg tooltips), and autohideables on the main workspace. Since the method is now a workspace instance method, it still hides global modals but hides autohideables on the workspace it is called on. The old behavior can be achieved by `Blockly.common.getMainWorkspace().hideChaff()`.

This PR makes Blockly.hideChaff() deprecated and passes through calls to the main workspace. It also updates core and demos to use the new method.

### CODE REVIEW

File: core/blockly.js [hunk 1]
```diff
@@ -230,23 +226,14 @@ Blockly.onContextMenu_ = function(e) {
...
-  });
-};
+// Add a getter for Blockly.hideChaff, for legacy reasons.
+Object.defineProperty(Blockly, 'hideChaff', {
+  get: function() {
+    Blockly.utils.deprecation.warn(
+        'Blockly.hideChaff', 'September 2021', 'September 2022');
+    return Blockly.common.getMainWorkspace().hideChaff;
+  }
+});
```
Comment (cpcallen) [inline core/blockly.js:236]: Why use an accessor here—wouldn't a stub wrapper suffice?:

```suggestion
/**
 * Close tooltips, context menus, dropdown selections, etc.
 * @deprecated Use Blockly.common.getMainWorkspace().hideChaff()
 * @param {boolean=} opt_onlyClosePopups Whether only popups should be closed.
 */
Blockly.hideChaff = function(opt_onlyClosePopups) {
  Blockly.utils.deprecation.warn(
      'Blockly.hideChaff', 'September 2021', 'September 2022');
  Blockly.common.getMainWorkspace().hideChaff(opt_onlyClosePopups);
};
```

This gives a place to hang the `@deprecation` notice and preserves type information.

File: core/shortcut_items.js [hunk 2]
```diff
@@ -47,8 +47,8 @@ const registerEscape = function() {
     preconditionFn: function(workspace) {
       return !workspace.options.readOnly;
     },
-    callback: function() {
-      Blockly.hideChaff();
+    callback: function(workspace) {
```
Comment (cpcallen) [inline core/shortcut_items.js:50]: You have verified that whatever calls this callback actually supplies a Workspace as the first argument?

File: core/theme_manager.js [hunk 3]
```diff
@@ -113,7 +113,9 @@ ThemeManager.prototype.setTheme = function(theme) {
     }
   }
 
-  Blockly.hideChaff();
+  for (let i = 0, workspace; (workspace = this.subscribedWorkspaces_[i]); i++) {
```
Comment (cpcallen) [inline core/theme_manager.js:116]: (Obligatory remark that the style guide says to prefer `for...of`.)

File: core/workspace_svg.js [hunk 4]
```diff
@@ -2328,7 +2331,7 @@ WorkspaceSvg.prototype.getScale = function() {
  * @package
  */
 WorkspaceSvg.prototype.scroll = function(x, y) {
-  Blockly.hideChaff(/* opt_onlyClosePopups */ true);
+  this.hideChaff(/* opt_onlyClosePopups */ true);
```
Comment (cpcallen) [inline core/workspace_svg.js:2334]: [The style guide recommends](https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments):

```suggestion
  this.hideChaff(/* opt_onlyClosePopups= */ true);
```

File: core/workspace_svg.js [hunk 5]
```diff
@@ -2644,4 +2647,21 @@ WorkspaceSvg.prototype.getGrid = function() {
...
+  Tooltip.hide();
+  WidgetDiv.hide();
+  DropDownDiv.hideWithoutAnimation();
+
+  var onlyClosePopups = !!opt_onlyClosePopups;
+  var autoHideables = this.getComponentManager().getComponents(
+      Blockly.ComponentManager.Capability.AUTOHIDEABLE, true);
+  autoHideables.forEach(function(autoHideable) {
+    autoHideable.autoHide(onlyClosePopups);
+  });
```
Comment (cpcallen) [inline core/workspace_svg.js:2664]: [Per style guide]:

```suggestion
  autoHideables.forEach((autoHideable) => autoHideable.autoHide(onlyClosePopups));
```
(Get clang-format to fix line length!)

**Review by cpcallen — APPROVED** [review]

File: core/shortcut_items.js (hunk 2 above)
Comment (gonfunko) [reply core/shortcut_items.js:50]: Yes, line 239 in shortcut_registry.js does this; the callbacks in this file just omit it (and the other args) if they don't care about them.

File: core/theme_manager.js (hunk 3 above)
Comment (gonfunko) [reply core/theme_manager.js]: Updated accordingly.

File: core/workspace_svg.js (hunk 4 above)
Comment (gonfunko) [reply core/workspace_svg.js]: Done!

File: core/workspace_svg.js (hunk 5 above)
Comment (gonfunko) [reply core/workspace_svg.js]: Done

**Pull request merged**